src/data/fintype/basic.lean

```diff
@@ -975,19 +976,13 @@ instance (α : Type*) [fintype α] : fintype (lex α) := ‹fintype α›
 @[simp] lemma fintype.card_lex (α : Type*) [fintype α] :
   fintype.card (lex α) = fintype.card α := rfl
 
-lemma univ_sum_type {α β : Type*} [fintype α] [fintype β] [fintype (α ⊕ β)] [decidable_eq (α ⊕ β)] :
-  (univ : finset (α ⊕ β)) = map function.embedding.inl univ ∪ map function.embedding.inr univ :=
-begin
-  rw [eq_comm, eq_univ_iff_forall], simp only [mem_union, mem_map, exists_prop, mem_univ, true_and],
-  rintro (x|y), exacts [or.inl ⟨x, rfl⟩, or.inr ⟨y, rfl⟩]
-end
-
 instance (α : Type u) (β : Type v) [fintype α] [fintype β] : fintype (α ⊕ β) :=
-@fintype.of_equiv _ _ (@sigma.fintype _
-    (λ b, cond b (ulift α) (ulift.{(max u v) v} β)) _
-    (λ b, by cases b; apply ulift.fintype))
-  ((equiv.sum_equiv_sigma_bool _ _).symm.trans
-    (equiv.sum_congr equiv.ulift equiv.ulift))
+{ elems := univ.disj_sum univ,
+  complete := by rintro (_ | _); simp }
+
+lemma univ_sum_type {α β : Type*} [fintype α] [fintype β] :
+  (univ : finset (α ⊕ β)) = univ.disj_sum univ :=
+rfl
```

Comment:
Can you provide this lemma turned around under the name `univ_disj_sum_univ` and then either keep or delete this one? The `_type` in the name is weird.